toddredmond/homework.md

```diff
@@ -0,0 +1,161 @@
+mean = (x+y+z) / 3;
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+mean = (x+y+z) / 3
+x = x+mea n;
+y = y+mean;
+z = z+mean;
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+area = (l+r)*(t+b)
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+(l+r)>(t+b)
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+area = (l+r)*(t+b)
+circumference = diameter*π
+circumference<=area or (diameter*π)<=((l+r)*(t+b))
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+((sqrt(a2+b2))*π)==((l+r)*(t+b))
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+var row, column;
+row = 2;
+column = 2;
+center = row && column;
+
+
+**4)** (_Difficulty: easyish_)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+* Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+* Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+**a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C. N = (R*8)+C
+
+**b)**  Given N, find R.  Write an expression for R which contains N. R = (N-R)/8
+
+**c)**  Given N, find C.  Write an expression for C which contains N. C = N-(R*8)
+
+**d)**  Assume the squares are colored black and white, with the upper-left square black.
+Write an expression to set a variable _color_ to either 'black' or 'white', describing the square identified by variables R,C, and N.
+(_Hint: If you prefer, you may use a pair of conditional statements instead of a single expression._)
+var counter = 0, var color;
+while (counter <= 63) {
+  if (counter%2===0) {
+    color = Black;
+  }
+  else {
+    color = White;
+  };
+  counter = counter + 1;
+}
+---
```

Comment:
For this one, what we're looking for here is something that will give the color for a particular square, rather than iterate through all of the squares. You're on the right track, though!